internal/backends/dart/dart.go

```diff
@@ -0,0 +1,250 @@
+}
+
+// dartListPubspecYaml lists all deps in a pubspec.yaml file
+func dartListPubspecYaml() map[api.PkgName]api.PkgSpec {
+
+	var specs dartPubspecYaml
+	specs = readSpecFile()
+
+	pkgs := map[api.PkgName]api.PkgSpec{}
+	for nameStr, specStr := range specs.Dependencies {
+		pkgs[api.PkgName(nameStr)] = api.PkgSpec(dartParseSpec(specStr))
+	}
+	for nameStr, specStr := range specs.DevDependencies {
+		pkgs[api.PkgName(nameStr)] = api.PkgSpec(dartParseSpec(specStr))
+	}
+	return pkgs
+}
+
+// ParseDartSpec parses a Dart version handling special
+// cases like Flutter SDK deps where the spec is a map.
+func dartParseSpec(spec interface{}) string {
+	version, ok := spec.(string)
+	// Handle cases like Flutter SDK deps where the spec is a map.
+	if !ok {
+		version = "Unknown"
+	}
+	return version
+}
+
+// PubspecLock represents deps in a pubspec.lock file.
+type dartPubspecLock struct {
+	Packages map[string]struct {
+		Dependency string `yaml:"dependency"`
+		Source     string `yaml:"source"`
+		Version    string `yaml:"version"`
+	} `yaml:"packages"`
+}
+
+// dartListPubspecLock lists all deps in a pubspec.lock file
+func dartListPubspecLock() map[api.PkgName]api.PkgVersion {
+	contentsB, err := ioutil.ReadFile("pubspec.lock")
+	if err != nil {
+		util.Die("pubspec.lock: %s", err)
+	}
+	var cfg dartPubspecLock
+	if err := yaml.Unmarshal(contentsB, &cfg); err != nil {
+		util.Die("pubspec.lock: %s", err)
+	}
+	pkgs := map[api.PkgName]api.PkgVersion{}
+	for nameStr, data := range cfg.Packages {
+		pkgs[api.PkgName(nameStr)] = api.PkgVersion(data.Version)
+	}
+	return pkgs
+}
+
+// pubDevSearchResults represents the data we get from Pub.dev when
+// doing a search.
+type pubDevSearchResults struct {
+	Packages []struct {
+		Name string `json:"package"`
+	} `json:"packages"`
+	NextURL string `json:"next"`
+}
+
+// dartSearch implements Search for Pub.dev.
+func dartSearch(query string) []api.PkgInfo {
+	endpoint := "https://pub.dev/api/search"
+	queryParams := "?q=" + url.QueryEscape(query)
+
+	resp, err := http.Get(endpoint + queryParams)
+	if err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+
+	var pubDevResults pubDevSearchResults
+	if err := json.Unmarshal(body, &pubDevResults); err != nil {
+		util.Die("Pub.dev: %s", err)
+	}
+
+	results := make([]api.PkgInfo, len(pubDevResults.Packages))
+	for i := range pubDevResults.Packages {
+		p := pubDevResults.Packages[i]
+		results[i] = api.PkgInfo{
+			Name: p.Name,
+		}
+	}
+	return results
+}
+
+// dartInfo is an empty stub implementation.
+func dartInfo(name api.PkgName) api.PkgInfo {
+	return api.PkgInfo{
+		Name:          "",
+		Description:   "",
+		Version:       "",
+		HomepageURL:   "",
+		SourceCodeURL: "",
+		BugTrackerURL: "",
+		Author: util.AuthorInfo{
+			Name:  "",
+			Email: "",
+			URL:   "",
+		}.String(),
+		License: ""}
+}
```

Comment:
If the API doesn't return this info, Name at least should be filled out so as to satisfy the UPM API contract.

Reply:
This is now implemented!